config_probe/key_not_found_within_config_probe_patterns.py

```diff
@@ -0,0 +1,2 @@
+class KeyNotFoundWithinConfigProbePatterns(AttributeError):
```

Comment:
2 things:

I'm not sure a module for that is relevant, putting it in the init is enough. especially since there's no body
The name seems a little overreaching.  ConfigNotFound could be good